pykrige/ok.py

```diff
@@ -194,44 +208,50 @@ class OrdinaryKriging:
                       'power': variogram_models.power_variogram_model,
                       'gaussian': variogram_models.gaussian_variogram_model,
                       'spherical': variogram_models.spherical_variogram_model,
-                      'exponential': variogram_models.exponential_variogram_model}
+                      'exponential': variogram_models.exponential_variogram_model,
+                      'hole-effect': variogram_models.hole_effect_variogram_model}
 
     def __init__(self, x, y, z, variogram_model='linear', variogram_parameters=None,
-                 variogram_function=None, nlags=6, weight=False, anisotropy_scaling=1.0,
-                 anisotropy_angle=0.0, verbose=False, enable_plotting=False,
-                 enable_statistics=False, coordinates_type='euclidean'):
+                 variogram_function=None, nlags=6, weight=False, use_psill=False,
+                 anisotropy_scaling=1.0, anisotropy_angle=0.0,
+                 verbose=False, enable_plotting=False, enable_statistics=False,
+                 coordinates_type='euclidean'):
 
-        # Code assumes 1D input arrays. Ensures that any extraneous dimensions
+        # Code assumes 1D input arrays of floats. Ensures that any extraneous dimensions
         # don't get in the way. Copies are created to avoid any problems with
         # referencing the original passed arguments.
-        self.X_ORIG = np.atleast_1d(np.squeeze(np.array(x, copy=True)))
-        self.Y_ORIG = np.atleast_1d(np.squeeze(np.array(y, copy=True)))
-        self.Z = np.atleast_1d(np.squeeze(np.array(z, copy=True)))
+        # Also, values are forced to be float... in the future, might be worth
+        # developing complex-number kriging (useful for vector field kriging)
+        self.X_ORIG = np.atleast_1d(np.squeeze(np.array(x, copy=True, dtype=np.float64)))
+        self.Y_ORIG = np.atleast_1d(np.squeeze(np.array(y, copy=True, dtype=np.float64)))
+        self.Z = np.atleast_1d(np.squeeze(np.array(z, copy=True, dtype=np.float64)))

```

Comment:
Cool. At some point in the future it would probably be worth putting in something to only convert to float64 if the input is not floating point. I can imagine forcing everything to float64 may create some problems for some people, since its got twice the memory requirement of float32...